test/over_react/component/fixtures/flawed_component.dart

```diff
@@ -32,28 +35,56 @@ class FlawedState extends _$FlawedState with _$FlawedStateAccessorsMixin {
 @Component()
 class FlawedComponent extends UiStatefulComponent<FlawedProps, FlawedState> {
```

Comment:
I know this matches the `3.1.0` branch implementation and was `Component` to begin with, but was there a reason it didn't get updated to `Component2` here?